src/app/controllers/frontend.server.controller.js

```diff
@@ -18,21 +18,27 @@ module.exports.datalayer = function (req, res) {
         'app_name': '<%= appName%>'
       });
     `
-  res
-    .type('text/javascript')
-    .status(StatusCodes.OK)
-    .send(ejs.render(js, req.app.locals))
+  try {
+    const ejsRendered = ejs.render(js, req.app.locals)
+    res.type('text/javascript').status(StatusCodes.OK).send(ejsRendered)
+  } catch (err) {
+    res.status(StatusCodes.BAD_REQUEST).json({ message: err })
```

Comment:
I'm not sure it's a good idea to just return the error message directly to the frontend. API error messages should be actionable for the user. when an error occurs, we should first log it, then return a more helpful message such as "There was an unexpected error. Please refresh and try again."

Reply:
edited